### PR TITLE
Feature/inverted length field

### PR DESCRIPTION
**Pull request by balo0801**
Implements the feature requested in #3  .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,13 +18,52 @@ before sending them over the serial interface.
...
 
-    +--------------+--------+-- - - --+-----+
-    | SYNC BYTE(S) | LENGTH | payload | CRC |
-    +--------------+--------+-- - - --+-----+
+    +--------------------+-----------------+--------------------------+-- - - --+--------------+
+    | SYNC BYTE [2 Byte] | LENGTH [2 Byte] | INVERTED LENGTH [2 Byte] | payload | CRC [2 Byte] |
+    +--------------------+-----------------+--------------------------+-- - - --+--------------+
+
+  - INVERTED LENGTH: Bit inverted length field.
+  - All other field are described in Serial MAC Protocol V1.
```
Comment (mrpelotazo) [inline README.md]: Fix field plural

File: src/sf_serialmac.h [hunk 2]
```diff
@@ -343,16 +348,19 @@ enum sf_serialmac_return sf_serialmac_rx_frame ( struct sf_serialmac_ctx *ctx,
...
-#define SF_SERIALMAC_PROTOCOL_SYNC_WORD              0xA5U
+#define SF_SERIALMAC_PROTOCOL_SYNC_WORD                  0xA5U
 /** Length of the STACKFORCE serial protocol SYNC word field. */
-#define SF_SERIALMAC_PROTOCOL_SYNC_WORD_LEN          0x01U
+#define SF_SERIALMAC_PROTOCOL_SYNC_WORD_LEN              0x01U
 /** Length of the STACKFORCE serial protocol length field */
-#define SF_SERIALMAC_PROTOCOL_LENGTH_FIELD_LEN       0x02U
+#define SF_SERIALMAC_PROTOCOL_LENGTH_FIELD_LEN           0x02U
+/** Length of the STACKFORCE serial protocol inverted length field */
+#define SF_SERIALMAC_PROTOCOL_LENGTH_INVERTED_FIELD_LEN  SF_SERIALMAC_PROTOCOL_LENGTH_FIELD_LEN
```
Comment (mrpelotazo) [inline src/sf_serialmac.h]: Length fields are the same length. The SF_SERIALMAC_PROTOCOL_LENGTH_INVERTED_FIELD_LEN define is redundant.

File: src/sf_serialmac.h [hunk 3]
```diff
@@ -343,16 +348,19 @@ enum sf_serialmac_return sf_serialmac_rx_frame ( struct sf_serialmac_ctx *ctx,
...
 /** Length of the STACKFORCE serial protocol CRC field */
-#define SF_SERIALMAC_PROTOCOL_CRC_FIELD_LEN          0x02U
+#define SF_SERIALMAC_PROTOCOL_CRC_FIELD_LEN              0x02U
 /** Length of the serial MAC frame header */
 #define SF_SERIALMAC_PROTOCOL_HEADER_LEN      \
- (SF_SERIALMAC_PROTOCOL_SYNC_WORD_LEN + SF_SERIALMAC_PROTOCOL_LENGTH_FIELD_LEN)
+ (SF_SERIALMAC_PROTOCOL_SYNC_WORD_LEN + SF_SERIALMAC_PROTOCOL_LENGTH_FIELD_LEN + \
+  SF_SERIALMAC_PROTOCOL_LENGTH_INVERTED_FIELD_LEN)
```
Comment (mrpelotazo) [inline src/sf_serialmac.h]: Replace redundant SF_SERIALMAC_PROTOCOL_LENGTH_INVERTED_FIELD_LEN with SF_SERIALMAC_PROTOCOL_LENGTH_FIELD_LEN.

File: src/sf_serialmac.c [hunk 4]
```diff
@@ -291,12 +291,34 @@ static enum sf_serialmac_return rx ( struct sf_serialmac_ctx *ctx,
...
-    /** Inform upper layer that there has been a frame header received */
-    ctx->rxFrame.state = SF_SERIALMAC_PAYLOAD;
-    ctx->rx_buffer_event ( ctx, NULL, ctx->rxFrame.remains );
+    /** Get the inverted length field */
+    invertedLength = UINT8_TO_UINT16 ( ctx->rxFrame.headerMemory +
+                              SF_SERIALMAC_PROTOCOL_SYNC_WORD_LEN +
+                              SF_SERIALMAC_PROTOCOL_LENGTH_INVERTED_FIELD_LEN);
+
+    /** Validate the header. */
+    if( ctx->rxFrame.remains == ~invertedLength )
```
Comment (mrpelotazo) [inline src/sf_serialmac.c]: Cast **~invertedLength** as uint16_t

File: README.md [hunk 5]
```diff
@@ -18,13 +18,52 @@ before sending them over the serial interface.
...
+
+### Serial MAC Protocol V1
+
+The Frame format is:
+
+    +--------------------+-----------------+-- - - --+--------------+
+    | SYNC BYTE [1 Byte] | LENGTH [2 Byte] | payload | CRC [2 Byte] |
+    +--------------------+-----------------+-- - - --+--------------+
+
+  - SYNC BYTE: static value 0xA5.
```
Comment (mrpelotazo) [inline README.md]: Write static with capital "S"

File: README.md [hunk 6]
```diff
@@ -18,13 +18,52 @@ before sending them over the serial interface.
...
+  - LENGTH: Amount of bytes transmitted with this serial frame. HEADER and CRC are not counted.
```
Comment (mrpelotazo) [inline README.md]: Transmitted payload size. HEADER and ...

File: README.md [hunk 7]
```diff
@@ -18,13 +18,52 @@ before sending them over the serial interface.
...
+
+The Serial MAC Protocol version describes the structure of a Serial MAC frame.
+Please note that this version is independent from the product version.
+
+### Serial MAC Protocol V1
+
+The Frame format is:
+
+    +--------------------+-----------------+-- - - --+--------------+
+    | SYNC BYTE [1 Byte] | LENGTH [2 Byte] | payload | CRC [2 Byte] |
```
Comment (mrpelotazo) [inline README.md]: Use capitals consequently in all field names.

File: README.md [hunk 8]
```diff
@@ -18,13 +18,52 @@ before sending them over the serial interface.
...
+
+### Serial MAC Protocol V2
 
 The Frame format is:
 
-    +--------------+--------+-- - - --+-----+
-    | SYNC BYTE(S) | LENGTH | payload | CRC |
-    +--------------+--------+-- - - --+-----+
+    +--------------------+-----------------+--------------------------+-- - - --+--------------+
+    | SYNC BYTE [2 Byte] | LENGTH [2 Byte] | INVERTED LENGTH [2 Byte] | payload | CRC [2 Byte] |
```
Comment (mrpelotazo) [inline README.md]: Use capitals consequently in all field names.

File: src/sf_serialmac.h (hunk 2 above)
Comment (balo0801) [reply src/sf_serialmac.h]: Yes you are right, this macro is not really needed.

File: README.md (hunk 7 above)
Comment (balo0801) [reply README.md]: - [x] Done.

File: README.md (hunk 5 above)
Comment (balo0801) [reply README.md]: - [x] Done.

File: README.md (hunk 6 above)
Comment (balo0801) [reply README.md]: - [x] Done.

File: README.md (hunk 8 above)
Comment (balo0801) [reply README.md]: - [x] Done.

File: README.md (hunk 1 above)
Comment (balo0801) [reply README.md]: - [x] Done.

File: src/sf_serialmac.c (hunk 4 above)
Comment (balo0801) [reply src/sf_serialmac.c]: - [x] Done.

File: src/sf_serialmac.h (hunk 2 above)
Comment (balo0801) [reply src/sf_serialmac.h]: - [x] Done.

File: src/sf_serialmac.h (hunk 3 above)
Comment (balo0801) [reply src/sf_serialmac.h]: - [x] Done.

File: README.md [hunk 9]
```diff
@@ -18,13 +18,52 @@ before sending them over the serial interface.
...
+
+### Serial MAC Protocol V1
+
+The Frame format is:
+
+    +--------------------+-----------------+-- - - --+--------------+
+    | SYNC BYTE [1 Byte] | LENGTH [2 Byte] | PAYLOAD | CRC [2 Byte] |
+    +--------------------+-----------------+-- - - --+--------------+
+
+  - SYNC BYTE: Static value 0xA5.
```
Comment (mrpelotazo) [inline README.md]: Please use **Fixed value** to avoid confusion with C's static.

Comment (balo0801) [reply README.md]: - [x] Done.

**Pull request merged**